codebuild/bin/install_al2_dependencies.sh

```diff
@@ -16,12 +16,50 @@
 set -eu
 source ./codebuild/bin/s2n_setup_env.sh
 
-if [[ ${DISTRO} != "amazon linux" ]]; then
-    echo "Target AL2, but running on $DISTRO: Nothing to do."
-    exit 0
-fi
+al2023_main(){
+    case "$S2N_LIBCRYPTO" in
+    "openssl-3.0") echo "Installing AL2023 packages";;
+    *) echo "${S2N_LIBCRYPTO} is not installed on this platform."; exit 1;;
+    esac
```

Comment:
Should this allow "default"? The "not installed on this platform" message might be confusing if "default" was requested.

Reply:
Yes, I'll add it as a synonym.